edocumentation-domain/src/main/java/org/dcsa/edocumentation/domain/persistence/entity/ShippingInstruction.java

```diff
@@ -224,6 +224,20 @@ public class ShippingInstruction extends AbstractStateMachine<EblDocumentStatus>
   @JoinColumn(name = "shipping_instruction_id")
   private List<@Valid AdvanceManifestFilingEBL> advanceManifestFilings;
 
+  @ElementCollection
+  @Column(name = "requested_certificate", nullable = false)
+  @CollectionTable(name = "requested_carrier_certificate", joinColumns = @JoinColumn(name = "shipping_instruction_id"))
+  @RequestedCarrierCertificateValidation( groups = AsyncShipperProvidedDataValidation.class)
+  @OrderColumn(name = "element_order")
+  private @Valid List<String> requestedCarrierCertificates;
+
+  @ElementCollection
+  @Column(name = "requested_clause", nullable = false)
+  @CollectionTable(name = "requested_carrier_clause", joinColumns = @JoinColumn(name = "shipping_instruction_id"))
+  @RequestedCarrierClauseValidation( groups = AsyncShipperProvidedDataValidation.class)
```

Comment:
Couldn't we just use `@PseudoEnum` here?

Reply:
`private List<@Valid @PseudoEnum(...) String> attributeName;`

(or something along those lines)